lib/routes/recovery-keys.js

```diff
@@ -0,0 +1,230 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const errors = require('../error')
+const isA = require('joi')
+const random = require('../crypto/random')
+const requestHelper = require('../routes/utils/request_helper')
+const validators = require('./validators')
+const HEX_STRING = validators.HEX_STRING
+const Promise = require('bluebird')
+
+module.exports = (log, db, Password, verifierVersion, customs) => {
+  return [
+    {
+      method: 'POST',
+      path: '/recoveryKeys',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: isA.string().max(32),
```

Comment:
Where does `recoveryKeyId` come from, is it surfaced in the UI? I would have assumed `recoveryKeyId` is generated by the server, so I'll assume I misunderstand its intent.

Reply:
In the auth server, are items optional by default or required by default? If optional, do these need to be required?

Reply:
Yea both these are required. `recoveryKeyId` is generated on the content-server and will be 

```javascript
hkdf(recoveryCode, uid, 'fxa recovery fingerprint', 16)
```

Doing this allows the auth-server no knowledge of the actual recovery code created.